warehouse/forklift/legacy.py

```diff
@@ -864,7 +864,9 @@ def file_upload(request):
                     lambda: request.POST["content"].file.read(8096), b""):
                 file_size += len(chunk)
                 if file_size > file_size_limit:
-                    raise _exc_with_message(HTTPBadRequest, "File too large.")
+                    raise _exc_with_message(HTTPBadRequest, 
+                       "File too large. Limit for project {} is {}.".format(
+                           project.name, file_size_limit))
```

Comment:
Note that `file_size_limit` is in bytes, so this is going to output something like:

> File is too large. Limit for project lambda_packages is 62914560.

This should probably be in MB, with a unit. It'd be nice if the project name was in quotes as well.